app/models/course.rb

```diff
@@ -386,23 +390,39 @@ def next_recruitment_cycle?
   # Ideally this would just use the validation, but:
   # https://github.com/rails/rails/issues/13971
   def course_params_assignable(course_params)
-    entry_requirements_assignable(course_params) &&
+    assignable_after_publish(course_params) &&
+      entry_requirements_assignable(course_params) &&
       qualification_assignable(course_params)
   end
 
 private
 
+  def assignable_after_publish(course_params)
+    relevant_params = course_params.slice(:is_send, :applications_open_from, :application_start_date)
+
+    invalid_params = relevant_params.select { is_published? }
```

Comment:
This pattern threw me ... you're selecting on something unrelated to any of the params being iterated through. I think a guard would be more appropriate, or at least something that more clearly expresses that if the course is published the list of invalid params is empty.